### DISCUSSION TITLE
via json

### DISCUSSION THREAD

**Discussion by sveeig** [manna-harbour/miryoku#74] [Q&A] [answered]
So I got a new Ready 2 Go corne 3x6 and want miryoku on there. It came out of the box VIA enabled, which is nice. 
However, it would be even nicer if I could just download VIA config json, load it into the VIA gui and had my miryoku setup without any compiling or flashing.

Via can also import external json layout files - I assume these are meant to be fro keyboard-layout-editor. That might also be a good option to provide miryoku to via enabled boards.

I had a look at the QMK json files which are provided in this repo and compared them to the via json format. The root structure is similar but unfortunately the order of keycodes within each layer is completely different. where QMK json has all keys per (complete) row in order, via goes left-to-right for each row on the left half first and then does the same right-to-left on the right hand board. There are also empty keys on the thumb row to "fill the grid", making it 48 codes per layer as opposed to qmk's 42 for the 6x3 corne.

**Comment by sveeig** [comment]
I just realized via only supports 4 layers. so this may not be possible at all

**Comment by manna-harbour** [comment, marked as answer]
I haven't looked into via / vial yet but plan on doing it at some point, see #54.

I believe you can just enable it at the keymap level and it'll import the keymap.  You'll need to increase the default layer limit.  See https://www.caniusevia.com/docs/configuring_qmk.

Doing it just by importing json would be less useful as it'd be tied to a single keyboard and it'd lack the necessary settings.  Also, the KLE json would not be compatible, and the QMK configurator json included here is no longer maintained and I imagine would also be incompatible.
